pkg/kubelet/kubelet_volumes.go

```diff
@@ -106,10 +106,14 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(
 		}
 		// If there are still volume directories, do not delete directory
 		volumePaths, err := kl.getPodVolumePathListFromDisk(uid)
-		if err != nil || len(volumePaths) > 0 {
+		if err != nil {
 			glog.Errorf("Orphaned pod %q found, but error %v occured during reading volume dir from disk", uid, err)
 			continue
 		}
+		if len(volumePaths) > 0 {
+			glog.Errorf("Orphaned pod %q found, but there are still %d directories under volumes, so fail to clean up", uid, len(volumePaths))
+			continue
+		}
```

Comment:
This mostly looks good. Just wanted to add that, fix doesn't work for plugins that don't use `getDeviceNameFromMount` such as `HostPath` . But may be we can open that as a separate issue.